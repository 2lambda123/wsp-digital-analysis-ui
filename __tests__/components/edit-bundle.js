/* global describe, it, expect, jest */

import { mount } from 'enzyme'
import { mountToJson } from 'enzyme-to-json'
import React from 'react'

import EditBundle from '../../lib/components/edit-bundle'

describe('Component > EditBundle', () => {
  it('renders correctly', () => {
    const props = {
      addBundle: jest.fn(),
      bundle: {},
      bundleId: '1',
      deleteBundle: jest.fn(),
      name: 'Test Bundle',
      projectId: '1',
      saveBundle: jest.fn()
    }

    // mount component
    const tree = mount(
      <EditBundle
        {...props}
        />
    )
    expect(mountToJson(tree)).toMatchSnapshot()
    const noCalls = [
      'addBundle',
      'deleteBundle',
      'saveBundle'
    ]
    noCalls.forEach((fn) => {
      expect(props[fn]).not.toBeCalled()
    })
  })
})
