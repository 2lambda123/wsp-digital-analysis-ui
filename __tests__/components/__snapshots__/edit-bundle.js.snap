exports[`Component > EditBundle renders correctly 1`] = `
<EditBundle
  addBundle={[Function]}
  bundle={Object {}}
  bundleId="1"
  deleteBundle={[Function]}
  name="Test Bundle"
  projectId="1"
  saveBundle={[Function]}>
  <pure(Panel)>
    <div
      className="panel panel-default">
      <pure(Heading)>
        <div
          className="panel-heading clearfix ">
          Edit bundle
        </div>
      </pure(Heading)>
      <pure(Body)>
        <div
          className="panel-body">
          <form
            encType="multipart/form-data"
            method="post"
            onSubmit={[Function]}>
            <pure(Component)
              name="Name"
              onChange={[Function]}
              placeholder="Bundle name"
              value="Test Bundle">
              <Group
                name="Name"
                onChange={[Function]}
                placeholder="Bundle name"
                value="Test Bundle">
                <div
                  className="form-group">
                  <pure(Component)
                    name="Name"
                    onChange={[Function]}
                    placeholder="Bundle name"
                    type="text"
                    value="Test Bundle">
                    <input
                      className="form-control"
                      defaultValue="Test Bundle"
                      name="Name"
                      onChange={[Function]}
                      placeholder="Bundle name"
                      type="text"
                      value="Test Bundle" />
                  </pure(Component)>
                </div>
              </Group>
            </pure(Component)>
            <input
              name="projectId"
              type="hidden"
              value="1" />
            <input
              className="btn btn-block btn-success"
              disabled={false}
              type="submit"
              value="Edit bundle" />
            <pure(Component)
              block={true}
              onClick={[Function]}
              style="danger">
              <a
                className="btn btn-danger btn-block"
                href="#"
                onClick={[Function]}
                target={undefined}>
                <pure(Icon)
                  type="trash">
                  <i
                    className="fa fa-trash fa-fw " />
                </pure(Icon)>
              </a>
            </pure(Component)>
          </form>
        </div>
      </pure(Body)>
    </div>
  </pure(Panel)>
</EditBundle>
`;
